### PR TITLE
[server] Add maptip for raster layer GetFeatureInfo

### LINKED ISSUE #55569: [QGIS Server] map tip content is missing for GetFeatureInfo response for WMS with GeoTIFF source
### What is the bug or the crash?

Map tip content is missing for GetFeatureInfo response for WMS with GeoTIFF source despite the parameter [WITH_MAPTIP=TRUE](https://docs.qgis.org/3.28/en/docs/server_manual/services/wms.html#with-maptip) is used:

image

### Steps to reproduce the issue

1. [test_project_and_data.zip](https://github.com/qgis/QGIS/files/13637315/test_project_and_data.zip)
2. Request:
`http://localhost:8000/?SERVICE=WMS&VERSION=1.3.0&REQUEST=GetFeatureInfo&BBOX=6.19864597499999981%2C51.21425923381612222%2C6.2280470250000004%2C51.27106176618387678&CRS=CRS%3A84&WIDTH=397&HEIGHT=767&LAYERS=test&STYLES=&FORMAT=image%2Fpng&QUERY_LAYERS=test&INFO_FORMAT=text%2Fhtml&I=198&J=398&WITH_MAPTIP=TRUE`

### Versions

QGIS version | 3.35.0-Master | QGIS code revision | 7af4536d0f6

### Supported QGIS version

- [X] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

For WMS with vector source the map tip content is included for GetFeatureInfo response.

### PR DESCRIPTION
**Pull request by pathmapper**
Since https://github.com/qgis/QGIS/pull/50854 raster maptips are available, but they are [currently not included](https://github.com/qgis/QGIS/pull/50854#issuecomment-1310309834) for the raster layer GetFeatureInfo response when requesting with parameter `WITH_MAPTIP=TRUE` like it is the case for vector layers.

This PR is a proposal to implement maptip for raster layer GetFeatureInfo.
In this context, the variable `layer_cursor_point` introduced with https://github.com/qgis/QGIS/pull/50854 holds the GetFeatureInfo position, so that maptips which are working in QGIS desktop are working also with QGIS Server GetFeatureInfo.

closes https://github.com/qgis/QGIS/issues/55569

### CODE REVIEW

File: src/server/services/wms/qgswmsrenderer.h [hunk 1]
```diff
@@ -297,6 +297,7 @@ namespace QgsWms
       bool featureInfoFromRasterLayer( QgsRasterLayer *layer,
                                        const QgsMapSettings &mapSettings,
                                        const QgsPointXY *infoPoint,
+                                       QgsRenderContext &renderContext,
```
Comment (m-kuhn) [inline src/server/services/wms/qgswmsrenderer.h:300]: Could we keep this const?

Comment (pathmapper) [reply src/server/services/wms/qgswmsrenderer.h]: Thanks, done in https://github.com/qgis/QGIS/pull/55680/commits/e7c43d004d3fd7fd3dbe487c921d665e56665b0c .

File: src/server/services/wms/qgswmsrenderer.cpp [hunk 2]
```diff
@@ -2307,6 +2308,19 @@ namespace QgsWms
...
         }
       }
+      //add maptip attribute based on html/expression
+      QString mapTip = layer->mapTipTemplate();
+      if ( !mapTip.isEmpty() && mWmsParameters.withMapTip() )
+      {
+        QDomElement maptipElem = infoDocument.createElement( QStringLiteral( "Attribute" ) );
+        maptipElem.setAttribute( QStringLiteral( "name" ), QStringLiteral( "maptip" ) );
+        QgsExpressionContext context { renderContext.expressionContext() };
+        QgsExpressionContextScope *scope = QgsExpressionContextUtils::layerScope( layer );
```
Comment (m-kuhn) [inline src/server/services/wms/qgswmsrenderer.cpp:2318]: Here we probably want globalProjectLayerScopes and add an extra scope for the additional variable.

**Comment by m-kuhn** [comment]
One more thing that skipped my attention in the first review, sorry for adding that now.
After that it's good for me, thanks :+1:

File: src/server/services/wms/qgswmsrenderer.cpp (hunk 2 above)
Comment (pathmapper) [reply src/server/services/wms/qgswmsrenderer.cpp]: done in https://github.com/qgis/QGIS/pull/55680/commits/b0e2e99b658c90546ea305eefdfcbdf5f0e3f419

**Comment by pathmapper** [comment]
Thank you for the review, much appreciated.

**Comment by m-kuhn** [comment]
You will probably question me ... my latest comment was completely wrong, I somehow missed that you were using the rendering context which was more appropriate :see_no_evil:

**Comment by pathmapper** [comment]
No worries, that means that your first comment was right :-) Thanks again :+1:

**Review by m-kuhn — APPROVED** [review]

**Comment by pathmapper** [comment]
Looks like there are currently two required status checks which will be pending forever and block merging (screenshot attached).

This is also the case for other PRs, e.g. https://github.com/qgis/QGIS/pull/55679, https://github.com/qgis/QGIS/pull/55700 .

It looks like these two checks were renamed and the old name is still required but the workflow with the old name doesn't exist anymore.

Maybe this could be resolved in the repo settings / branch protection rules -> https://github.com/orgs/community/discussions/26698

image

**Comment by pathmapper** [comment]
Seems like this is the commit which causes this (additional `OFF` in the name): https://github.com/qgis/QGIS/commit/d04355ca0da1cf4e22171d8bc8f902dc1cf15717

**Comment by m-kuhn** [comment]
Thanks for the analysis, I noticed but didn't jump into the rabbit hole. This should be fixed now.

**Pull request merged**